test/jdk/java/foreign/TestByteBuffer.java

```diff
@@ -290,26 +285,62 @@ public void testMappedSegmentOffset() throws Throwable {
 
         // write one at a time
         for (int i = 0 ; i < tuples.byteSize() ; i += tupleLayout.byteSize()) {
-            try (Arena arena = Arena.openConfined();
+            try (Arena arena = Arena.ofConfined();
                  FileChannel fileChannel = FileChannel.open(f.toPath(), StandardOpenOption.READ, StandardOpenOption.WRITE)) {
                 //write to channel
-                MemorySegment segment = fileChannel.map(FileChannel.MapMode.READ_WRITE, i, tuples.byteSize(), arena.scope());
+                MemorySegment segment = fileChannel.map(FileChannel.MapMode.READ_WRITE, i, tuples.byteSize(), arena);
                 initTuples(segment, 1);
                 segment.force();
             }
         }
 
         // check one at a time
         for (int i = 0 ; i < tuples.byteSize() ; i += tupleLayout.byteSize()) {
-            try (Arena arena = Arena.openConfined();
+            try (Arena arena = Arena.ofConfined();
                  FileChannel fileChannel = FileChannel.open(f.toPath(), StandardOpenOption.READ)) {
                 //read from channel
-                MemorySegment segment = fileChannel.map(FileChannel.MapMode.READ_ONLY, 0L, tuples.byteSize(), arena.scope());
+                MemorySegment segment = fileChannel.map(FileChannel.MapMode.READ_ONLY, 0L, tuples.byteSize(), arena);
                 checkTuples(segment, segment.asByteBuffer(), 1);
             }
         }
     }
 
+    @Test(dataProvider = "fromArrays")
+    public void testAsByteBufferFromNonByteArray(MemorySegment segment) {
+        if (!segment.heapBase().map(a -> a instanceof byte[]).get()) {
+            // This should not work as the segment is not backed by a byte array
+            assertThrows(UnsupportedOperationException.class, segment::asByteBuffer);
+        }
+    }
+
+    @Test
+    public void testMappedSegmentAsByteBuffer() throws Throwable {
```

Comment:
While testing something else, I noticed that this test is failing on Linux/WSL since the WSL 1 kernel does not implement `mincore`. We can simply skip the test in that case, as we already do for other tests. I've submitted a patch here: https://github.com/minborg/jdk/pull/2

Reply:
Thanks for providing this patch @JornVernee